### PR TITLE
[TS] LPS-131894

### PR DESCRIPTION
**Pull request by ericyanLr**
Hi @tinatian ,

Can you help review this PR?
https://issues.liferay.com/browse/LPS-131894

Notes from @nikolettabuza:

> `object` can be a proxy instance and in that case, the `function.apply(object)` throws a `ClassCastException`.
> As we discussed, I replaced this approach with `baseModel.getModelAttributes()` to access the `companyId`.
> 
> I aslo updated the `RestrictedLiferayObjectWrapperTest` since this new logic requires the `getModelAttributes()` to return the `companyId` through `getAttributeGetterFunctions()`.

I also provide additional context below, in case its helpful.

Please let us know if you have any questions.
Thanks!

----

**Additional Context from PTR-2416**

> I took a look at the issue and it looks like the logic involved (RestrictedLiferayObjectWrapper.java#L221-L229), cannot handle our ServiceWrappers that utilize ModelAdapterUtil, such as AssetCategoryPropertyLocalServiceWrapper.
> 
> As you probably noticed, the object is a ProxyClass and the associated InvocationHandler is the DelegateInvocationHandler that's provided by ModelAdapterUtil.

### CODE REVIEW

**Comment by ericyanLr** [comment]
ci:test:sf

**Comment by ericyanLr** [comment]
ci:test:relevant

File: modules/apps/portal-template/portal-template-freemarker/src/main/java/com/liferay/portal/template/freemarker/internal/RestrictedLiferayObjectWrapper.java [hunk 1]
```diff
@@ -218,15 +217,12 @@ else if (object instanceof BaseModel) {
 			if (currentCompanyId != CompanyConstants.SYSTEM) {
 				BaseModel<?> baseModel = (BaseModel<?>)object;
 
-				Map<String, Function<Object, Object>> getterFunctions =
-					(Map<String, Function<Object, Object>>)
-						(Map<String, ?>)baseModel.getAttributeGetterFunctions();
+				Map<String, Object> modelAttributes =
+					baseModel.getModelAttributes();
```
Comment (tinatian) [inline modules/apps/portal-template/portal-template-freemarker/src/main/java/com/liferay/portal/template/freemarker/internal/RestrictedLiferayObjectWrapper.java:221]: @ericyanLr @nikolettabuza The reason we did not use "baseModel.getModelAttributes();" is it is too heavy, it need to calculate the values for all attributes, and then only use one of hem (companyId). I will take a look at this issue next week.

Tina.

**Comment by tinatian** [comment]
@ericyanLr @nikolettabuza I am going to close this pull, please see my comments on the ticket here https://issues.liferay.com/browse/LPS-131894?focusedCommentId=2442939&page=com.atlassian.jira.plugin.system.issuetabpanels:comment-tabpanel#comment-2442939.

**Pull request closed without merging**